### DISCUSSION TITLE
(Solve) Error during installation

### DISCUSSION THREAD

**Discussion by ireneegomezz** [chris1111/Wireless-USB-Big-Sur-Adapter#52] [Q&A]
Hello,

Since I downloaded Big Sur, my cudy wireless USB adapter isn't working. I tried to install this driver, but I can't because when it is almost installed, it shows the same error over and over again so the installation is never completed.

Can anybody help me, please?

<img width="623" alt="Captura de pantalla 2021-01-07 a las 10 21 16" src="https://user-images.githubusercontent.com/77103952/103887304-1f6d7f00-50e3-11eb-8e38-dafd47da8a2e.png">

**Comment by ireneegomezz** [comment]
Sorry, all I had to do was disable SIP!! Now it works wonderfully :) I guess now it is installed I can enable again SIP?

**Comment by chris1111** [comment]
SIP Must stay disable

**Reply by ireneegomezz** [reply]
SIP disable could lead to any problems like virus or privacy intruders?

**Comment by chris1111** [comment]
The mechanism of System Integrity Protection (SIP), or "protection of the integrity of the system" in the language of Doomams, limits the powers of the root user. But the root user is supposed to ... have all the powers. SIP therefore disrupts the installation and operation of utilities accessing the depths of the system. In this case, it may be useful to deactivate it.

While some may think it is, SIP was not designed to make your life difficult. It actually protects you from attacks against the biggest flaw in how macOS works: you. The password for the administrator account allows you to "borrow" the rights of the root account, and even change the password, and thus have full permissions on the system.

But "most Macs are used by one person," says Apple, a person "who has administrative rights by default". Give your administrator password to a malicious application, like a Trojan horse masquerading as a benign application, and you will give it control over your machine (read: How SIP Completes Apple's Security Policy)

Disabling SIP will not cause immediate disaster. After all, SIP didn't exist before OS X El Capitan, and we were doing pretty well

**Reply by ireneegomezz** [reply]
Chris, thank you very much for your quick and clarifying answer. I guess then I shouldn't have any problems and I can always enable it again when cudy launches their own driver.

Thanks again,

Irene
